Library/ViewModels/RewardAddOnSelectionViewModelTests.swift

```diff
@@ -1107,8 +1107,8 @@ final class RewardAddOnSelectionViewModelTests: TestCase {
     self.vm.inputs.configure(with: data)
     self.vm.inputs.viewDidLoad()
 
-    XCTAssertEqual(["Add-Ons Page Viewed"], self.dataLakeTrackingClient.events)
-    XCTAssertEqual(["Add-Ons Page Viewed"], self.segmentTrackingClient.events)
+    XCTAssertEqual(["CTA Clicked"], self.dataLakeTrackingClient.events)
```

Comment:
A Page Viewed event should not be translating to a CTA Clicked. There is a good chance some of the logic you introduced to the view model is creating this issue, given it's occurring in multiple places throughout the file. Double check this.